drivers/net/w5500.c

```diff
@@ -0,0 +1,2232 @@
+
+/* Socket n Command Register (SN_CR) */
+
+#define SN_CR_OPEN           0x01      /* Socket n is initialized and opened according
+                                        * to the protocol selected in SN_MR */
+#define SN_CR_LISTEN         0x02      /* Socket n operates as a 'TCP server' and waits
+                                        * for connection request from any 'TCP client' */
+#define SN_CR_CONNECT        0x04      /* 'TCP client' connection request */
+#define SN_CR_DISCON         0x08      /* TCP disconnection request */
+#define SN_CR_CLOSE          0x10      /* Close socket n */
+#define SN_CR_SEND           0x20      /* Transmit all data in Socket n TX buffer */
+#define SN_CR_SEND_MAC       0x21      /* Transmit all UDP data (no ARP) */
+#define SN_CR_SEND_KEEP      0x22      /* Send TCP keep-alive packet */
+#define SN_CR_RECV           0x40      /* Complete received data in Socket n RX buffer */
+
+/* Socket n Interrupt Register (SN_IR) and
+ * Socket n Interrupt Mask Register (SN_IMR)
+ */
+
+#define SN_INT_CON           (1 << 0)  /* Bit 0:  Connection with peer successful */
+#define SN_INT_DISCON        (1 << 1)  /* Bit 1:  FIN or FIN/ACK received from peer */
+#define SN_INT_RECV          (1 << 2)  /* Bit 2:  Data received from peer */
+#define SN_INT_TIMEOUT       (1 << 3)  /* Bit 3:  ARP or TCP timeout */
+#define SN_INT_SEND_OK       (1 << 4)  /* Bit 4:  SEND command completed */
+
+/* Socket n Status Register (SN_SR) */
+
+#define SN_SR_SOCK_CLOSED      0x00
+#define SN_SR_SOCK_INIT        0x13
+#define SN_SR_SOCK_LISTEN      0x14
+#define SN_SR_SOCK_ESTABLISHED 0x17
+#define SN_SR_SOCK_CLOSE_WAIT  0x1c
+#define SN_SR_SOCK_UDP         0x22
+#define SN_SR_SOCK_MACRAW      0x42
+
+#define SN_SR_SOCK_SYNSENT     0x15    /* Transitional status */
+#define SN_SR_SOCK_SYNRECV     0x16
+#define SN_SR_SOCK_FIN_WAIT    0x18
+#define SN_SR_SOCK_CLOSING     0x1a
+#define SN_SR_SOCK_TIME_WAIT   0x1b
+#define SN_SR_SOCK_LAST_ACK    0x1d
+
+/* Socket n RX Buffer Size Register (SN_RXBUF) */
+
+#define SN_RXBUF_0KB          0
+#define SN_RXBUF_1KB          1
+#define SN_RXBUF_2KB          2
+#define SN_RXBUF_4KB          4
+#define SN_RXBUF_8KB          5
+#define SN_RXBUF_16KB         16
+
+/* Socket n TX Buffer Size Register (SN_TXBUF) */
+
+#define SN_TXBUF_0KB          0
+#define SN_TXBUF_1KB          1
+#define SN_TXBUF_2KB          2
+#define SN_TXBUF_4KB          4
+#define SN_TXBUF_8KB          5
+#define SN_TXBUF_16KB         16
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/* The w5500_driver_s encapsulates all state information for a single
+ * hardware interface
+ */
+
+struct w5500_driver_s
+{
+  bool sk_bifup;               /* true:ifup false:ifdown */
+  struct wdog_s sk_txtimeout;  /* TX timeout timer */
```

Comment:
Minor: I think it is good to remove `sk_` or change to something else since `sk_` is for `skeleton` code example.